package.json

```
@@ -10,32 +10,68 @@
     "semantic-release": "^17.1.1"
   },
   "release": {
-    "branches": ["main", {"name": "rc", "prerelease":  true}],
+    "branches": [
+      "main",
+      {
+        "name": "rc",
+        "prerelease": true
+      }
+    ],
     "plugins": [
-      ["@semantic-release/commit-analyzer",{
-        "preset": "eslint",
-        "releaseRules": [
-          {"tag": "Breaking", "release": "minor"},
-          {"tag": "Update", "release": "patch"},
-          {"tag": "Fix", "release": "patch"},
-          {"tag": "New", "release": "patch"},
-          {"tag": "Upgrade", "release": "patch"}
-        ]
-      }],
-      ["@semantic-release/release-notes-generator", {
-        "preset": "eslint"
-      }],
-      ["@semantic-release/exec",{
-        "verifyConditionsCmd": "semantic-release-rust verify-conditions",
-        "prepareCmd": "semantic-release-rust prepare ${nextRelease.version}",
-        "publishCmd": "semantic-release-rust publish"
-      }],
+      [
+        "@semantic-release/commit-analyzer",
+        {
+          "preset": "eslint",
+          "releaseRules": [
+            {
+              "tag": "Breaking",
+              "release": "minor"
+            },
+            {
+              "tag": "Update",
+              "release": "patch"
+            },
+            {
+              "tag": "Fix",
+              "release": "patch"
+            },
+            {
+              "tag": "New",
+              "release": "patch"
+            },
+            {
+              "tag": "Upgrade",
+              "release": "patch"
+            }
+          ]
+        }
+      ],
+      [
+        "@semantic-release/release-notes-generator",
+        {
+          "preset": "eslint"
+        }
+      ],
+      [
+        "@semantic-release/exec",
+        {
+          "verifyConditionsCmd": "semantic-release-rust verify-conditions",
+          "prepareCmd": "semantic-release-rust prepare ${nextRelease.version}",
```

This is almost-definitely what is bumping the version in `Cargo.toml`. Maybe `semantic-release-rust` has an option for updating the `.lock` file? If not, this is just a bash script, so we could do this:

```suggestion
          "prepareCmd": "semantic-release-rust prepare ${nextRelease.version} && cargo update -w",
```

Or, as we've done in all [our Python projects](https://github.com/rigetti/pyquil/blob/497791e8108d8780109d75410be786c5f6e590ea/package.json#L29) create a `prepare_release.sh` and invoke that here.

---

Good call -- I'm good inlining this at first and extracting out to a script if it gets more complex down the line. Thanks!

---

Looks like `semantic-release-rust` only touches the `Cargo.toml`, not the lockfile. What's more, the documentation has a number of unfortunate typos 😬

So! I support the `&& cargo update -w` maneuver!